### PR TITLE
Issue 20

### LINKED ISSUE #20: escape Path key with URL (encoding)
When using the "move to trash" from the GUI in XFCE, it seems that when deleting a file with spaces in it, a %20 will replace the space when it writes the filename to .trashinfo.

If rmw restores it, it will have a %20 replacing the space in it's name.

### PR DESCRIPTION
**Pull request by blah1898**
rmw now escapes paths in .trashinfo and is capable of reading as well. Fixes #20 as well.

### CODE REVIEW

**Comment by andy5995** [comment]
Thank you, @blah1898 It's late here for me, and I'm tired, but I skimmed over your code. I'll take a closer look tomorrow, resolve the merge conflicts, and patch my local copy to test it out. Have a great day (or night).

File: src/main.c [hunk 1]
```diff
@@ -302,7 +286,24 @@ main (int argc, char *argv[])
                               time_now, time_str_appended, current_waste_num);
 
             if (info_status == 0)
+            {
```
Comment (andy5995) [inline src/main.c]: Clearly these edits shouldn't be here. ;) I merged that patch in a few days ago. I'd suggest having a look at this [contributing page](https://github.com/andy5995/rmw/wiki/Contributing) to see how to avoid this - and other conflicts - in the future. Please don't hesitate to let me know if you have any questions.

File: src/primary_funcs.c [hunk 2]
```diff
@@ -99,9 +99,16 @@ mkinfo (struct rmw_target file, struct waste_containers *waste, char *time_now,
 
   char real_path[MP];
 
+  /* Worst case scenario: whole path is escaped, so 3 chars per
+   * actual character*/
+  char escaped_path[MP * 3];
```
Comment (andy5995) [inline src/primary_funcs.c]: Good thinking!

File: src/main.c (hunk 1 above)
Comment (blah1898) [inline src/main.c]: Yeah, sorry about that, I'm pretty bad at git (never used it for colaborating before). So, what I did here (in my local repo) was the following:

- Merge local develop with upstream develop
- Revert the commit that shouldn't be there
- Merge issue_20 with develop, fixing the conflicts

Should this fix it? If so, I'll push it and do another pull request.

Comment (andy5995) [inline src/main.c]: Yes, please do another pull request and let's have a look. :)

I haven't examined your patch yet fully, but I suspect that it will need to be integrated into purge_rmw.c as well.

Comment (blah1898) [inline src/main.c]: Hold on a second, I accidentally removed my last commit when messing around with history 😅. Will push in a few seconds.

Comment (andy5995) [inline src/main.c]: Ok, no worries.

I'm on Quakenet IRC in #rmw you want to stay in touch through this process.

**Comment by andy5995** [comment]
When I tested, the only problem I saw was that it doesn't work with my new patch for #14. Would you be interested in addressing that?

```
rmw -z "a a a"

Searching using only the basename...

/home/andy/.local/share/Trash/files/a a a not found

/home/andy/.trash.rmw/files/a a a not found

search complete

andy@oceanus:~/temp$ cd .local/share/Trash.test/files/

andy@oceanus:~/temp/.local/share/Trash.test/files$ rmw -z "a a a"

+'a a a' -> '/home/andy/temp/a a a'

```

**Comment by andy5995** [comment]
My mistake. I was using the rmw from my PATH, not the current build. Your patch is working as intended. Nice!
